sys/mips/trap.c

```diff
@@ -277,4 +282,6 @@ __no_profile void mips_exc_handler(ctx_t *ctx) {
   } else {
     intr_root_handler(ctx);
   }
+
+  td->td_kframe = kframe_saved;
```

Comment:
As far as I understood, previously we did not restore the old context value, why we didn't notice that?

Reply:
Because nobody relies on that (see e.g. how interrupts are handled - `intr_root_handler` takes the context as an argument although it could rely on the pointer in thread structure).